src/routes/conversation/[id]/+server.ts

```diff
@@ -410,6 +429,173 @@ export async function POST({ request, locals, params, getClientAddress }) {
 				}
 			}
 
+			const endpoint = await model.getEndpoint();
+			let toolResults: ToolResult[] | undefined = undefined;
+
+			// function calls, if no assistant or assistant has websearch enabled
+			if (model.functions && (!assistant || assistantHasWebSearch)) {
+				let tools: BackendTool[] = [];
+
+				// if it's an assistant, only support websearch for now
+				if (!assistant) {
+					// filter based on tool preferences, add the tools that are on by default
+					tools = [directlyAnswer, text2img, calculator, fetchUrl, websearch].filter((el) => {
+						if (el.isLocked && el.isOnByDefault) {
+							return true;
+						}
+
+						return toolsPreferences?.[el.name] ?? el.isOnByDefault;
+					});
```

Comment:
```suggestion
					tools = [directlyAnswer, text2img, calculator, fetchUrl, websearch].filter((el) => (el.isLocked && el.isOnByDefault) || toolsPreferences?.[el.name] ?? el.isOnByDefault);
```

wdyt? maybe not much readable ?